src/components/projects/wbs-details/work-package-container/dependencies-list/dependencies-list.tsx

```diff
@@ -7,22 +7,57 @@ import HorizontalList from '../../../../shared/horizontal-list/horizontal-list';
 import Dependency from './dependency/dependency';
 import './dependencies-list.module.css';
 import { Button, InputGroup, Form } from 'react-bootstrap';
-import { EditModeContext } from '../work-package-container';
-import { useContext } from 'react';
+import { FormContext } from '../work-package-container';
+import { useContext, useEffect, useState } from 'react';
+import { validateWBS } from '../../../../../utils/src/validate-wbs';
 
 interface DependenciesListProps {
   dependencies: WbsNumber[];
 }
 
 const DependenciesList: React.FC<DependenciesListProps> = ({ dependencies }) => {
-  const editMode = useContext(EditModeContext);
+  const { editMode } = useContext(FormContext);
+  const [dependenciesState, setDependenciesState] = useState(dependencies);
+  const [unvalidatedDependency, setUnvalidatedDependency] = useState('');
+
+  useEffect(() => {
+    setDependenciesState(dependencies);
+  }, [editMode, dependencies])
+  
   const AddButton = (
     <InputGroup>
-      <Form.Control type="text" placeholder="New WBS #"></Form.Control>
-      <Button variant="success">+</Button>
+      <Form.Control
+        type="text"
+        placeholder="New WBS #"
+        onChange={(e) => setUnvalidatedDependency(e.target.value)}
+      ></Form.Control>
+      <Button variant="success" onClick={handleAdd}>
+        +
+      </Button>
     </InputGroup>
   );
-  const items = dependencies.map((ele) => <Dependency wbsNumber={ele} />);
+
+  function handleDelete(dependencyToDelete: WbsNumber) {
+    const index = dependenciesState.indexOf(dependencyToDelete);
+    if (index !== -1) {
+      const half1 = dependenciesState.slice(0, index);
+      const half2 = dependenciesState.slice(index + 1);
+      setDependenciesState(half1.concat(half2));
+    }
+  }
+
+  function handleAdd() {
+    try {
+      setDependenciesState([...dependenciesState, validateWBS(unvalidatedDependency)]);
+    } catch (error: any) {
+      alert(error.message);
+    }
+    setUnvalidatedDependency('');
+  }
```

Comment:
@jamescd18 correct me if I'm wrong, but I think we generally try to use ES6 arrow function expressions over function declarations